File: Docker/waitfor.py
#!/usr/bin/python

import socket
import time
import os
import sys


addr, port = sys.argv[1], sys.argv[2]
port = int(port)

s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
while True:
    try:
        s.connect((addr, port))
        s.close()
        break
    except socket.error as ex:
        time.sleep(.5)

File: Docker/waitfor.pl
#!/usr/bin/perl
use strict;
use warnings;
use Socket qw(PF_INET SOCK_STREAM pack_sockaddr_in inet_aton);
use IO::Socket::INET;

my ( $addr, $port ) = @ARGV;

while (1) {
    socket( my $socket, PF_INET, SOCK_STREAM, 0 )
      or die "socket: $!";
    if ( connect( $socket, pack_sockaddr_in( $port, inet_aton($addr) ) ) ) {
        close $socket;
        exit 0;
    }
    else { sleep(1); }
}
